sound/soc/sof/intel/hda-ipc.c

```diff
@@ -226,10 +222,17 @@ irqreturn_t hda_dsp_ipc_irq_thread(int irq, void *context)
 
 		hda_dsp_ipc_host_done(sdev);
 
-		ret = IRQ_HANDLED;
+		ipc_irq = true;
 	}
 
-	return ret;
+	if (!ipc_irq)
+		return IRQ_NONE;
```

Comment:
Go without re-enable IPC interrupt? This looks NOK for me.

Reply:
Possibly, but then the SKL driver has an issue. The goal is to align and identify problematic flows, and it could be that SOF was correct and SKL not.